### PR TITLE
Fix #542 - set()

### LINKED ISSUE #542: Fix set()
There is currently a method called `set` that lives in `/lib/set.js`.
It is incomplete and needs to be fixed!

It basically behaves just like lodash's `set` method.
You can use it like this: 
```js
set({}, 'a.b.c', 123) // Returns {a: {b: {c: 123}}}

set({ hello: {world: 123}}, 'hello.kitty', 42)
/*
 Returns:
{
  hello: {
    world: 123,
  },
  kitty: 42,
}
*/
```

⚠️  Please wait for https://github.com/BlakeGuilloud/ganon/issues/537 to be merged before attempting this one.

Instead of specifying the path with a dotted string, you can also use an array (`['a', 'b', 'c']` for instance)

### Acceptance Criteria:
1. Running `yarn test set` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by SamHowie**
Closes #542 

- [x] Running `yarn lint` does not trigger any linter errors
- [x] The test of the method you have fixed is passing
- [x] You have written a new skeleton method for someone else to work on!
- [x] You have written tests to accompany your skeleton method

### CODE REVIEW

File: lib/set.js [hunk 1]
```diff
@@ -6,8 +6,36 @@
...
+    if (node[key] === undefined) {
+      node[key] = {};
+    }
+    node = node[key];
+  }
+
+  key = path[i];
+  node[key] = value;
+
+  return obj;
```
Comment (BlakeGuilloud) [inline lib/set.js]: Could you maybe explain how this is working? I'm a little confused that since you are returning the original `obj`parameter without mutating it. thanks!

Comment (SamHowie) [reply lib/set.js]: Sure! Mutation is actually being applied to the obj parameter.

See L23 where obj is assigned to the `node` variable.

We then walk the path updating the node variable to be the current point in the path. If the current node doesn't have that point in the path, we insert it.

Finally, for the final point in the path, we assign the value.

Comment (BlakeGuilloud) [reply lib/set.js]: Ah! Very cool. Thanks @SamHowie !! Gonna go ahead and merge this in. Thanks!
